### PR TITLE
fix solr with openjdk, stop patching paths in solr config files, drop support for broken examples

### PR DESCRIPTION
**Pull request by mkalkbrenner**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by mkalkbrenner** [comment]
jenkins: The URL https://lucene.apache.org/solr/ is not reachable (HTTP status code 404)

how could the test be restarted?

**Comment by SMillerDev** [comment]
@BrewTestBot test this please

**Comment by Bo98** [comment]
The audit is correct. The Solr website appears to have gone missing.

**Comment by SMillerDev** [comment]
Back now

**Comment by SMillerDev** [comment]
@Brewtestbot test this please

**Comment by mkalkbrenner** [comment]
@BrewTestBot test this please

**Comment by SMillerDev** [comment]
You can ignore the github action failures. Could you extend the test block to make sure this isn't broken again though? Maybe something akin to httpd?

**Comment by mkalkbrenner** [comment]
> Could you extend the test block to make sure this isn't broken again though?
Sure.
My idea is to query Solr's API after `solr start` to get its state.
But could we do it in a second step and get it back to work first by accepting the PR as it is?

Another thing that is missing is that the zookeeper scripts aren't handled at the moment. They need to be linked in `bin` if you want to start Solr in "cloud mode". But I wanted to avoid to introduce new features in this PR and start anotherone instead.

**Comment by mkalkbrenner** [comment]
Are there good examples of tests? Are there other formulas you could recommend as an example?

**Comment by SMillerDev** [comment]
https://github.com/Homebrew/homebrew-core/blob/master/Formula/httpd.rb has a good test. I'd prefer to improve the tests in this pull requests, to avoid accidentally breaking something in further PRs.

**Comment by mkalkbrenner** [comment]
I found two more bugs in the formulas while working on the tests:
The logs and the pid files were written to the Cellar instead of /usr/local/var.

Locally `brew test` passes sucessfully. But I don't know why they don't pass here. Any ideas?

**Comment by Bo98** [comment]
Looks like `solr` fails if the `/usr/local/var/run` directory doesn't exist.

File: Formula/solr.rb [hunk 1]
```diff
@@ -61,6 +50,10 @@ def plist
   end
 
   test do
-    system bin/"solr"
+    # Start a Solr node => exit code 0
+    shell_output(bin/"solr start")
+    sleep 10
```
Comment (SMillerDev) [inline Formula/solr.rb]: Could you make a query to solr here?

Comment (mkalkbrenner) [reply Formula/solr.rb]: yes, already added the queries in previous commit but removed them temporarily

Comment (SMillerDev) [reply Formula/solr.rb]: :+1: ignore this then

**Comment by mkalkbrenner** [comment]
The remaining issue is:
Parent for temp dir not configured correctly

This only seem to happen in brew test context.

**Comment by SMillerDev** [comment]
https://github.com/Homebrew/homebrew-core/blob/master/Formula/orientdb.rb#L37-L39 this could be a solution

**Comment by SMillerDev** [comment]
Looks good, just need to fix the `brew style solr` thing and then we can merge :tada:

**Comment by SMillerDev** [comment]
Okay, now for the final change. Can you squash all your commits into one?

**Comment by mkalkbrenner** [comment]
Isn't squash an option you could select during merge?
I think that is a github feature

**Comment by mkalkbrenner** [comment]
That went wrong :-(
Please merge #50759 instead.

**Pull request closed without merging**